### PR TITLE
Creating configuration metadata for bindings properties

### LINKED ISSUE #30: [SOL-57954] Enable IDE Awareness of Configuration Metadata
Enhancement request to allow for developers to use autocomplete and IDEs to be aware of Solace config properties when setting properties for the Spring Cloud Stream Binder. 

The path forward on this is likely using `@ConfigurationProperties` to specify classes that provide the config options and includes the `spring-boot-configuration-processor` dependency.  More info [here](https://www.baeldung.com/spring-boot-configuration-metadata)
```
<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-configuration-processor</artifactId>
    <optional>true</optional>
</dependency>
```

Consider also doing the same for other solace spring-boot-starters...

### PR DESCRIPTION
**Pull request by carolmorneau**
closes #30

### CODE REVIEW

File: solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceProducerProperties.java [hunk 1]
```diff
@@ -1,13 +1,27 @@
 package com.solace.spring.cloud.stream.binder.properties;
 
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+@ConfigurationProperties("spring.cloud.stream.solace.default.producer")
```
Comment (Nephery) [inline solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceProducerProperties.java:10]: See review comment for `SolaceConsumerProperties`.

File: solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceConsumerProperties.java [hunk 2]
```diff
@@ -1,32 +1,92 @@
 package com.solace.spring.cloud.stream.binder.properties;
 
 import com.solacesystems.jcsmp.EndpointProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 
 import java.util.concurrent.TimeUnit;
 
+@ConfigurationProperties("spring.cloud.stream.solace.default.consumer")
```
Comment (Nephery) [inline solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceConsumerProperties.java:8]: * This raises a compiler warning that the configuration isn't registered anywhere. Add a `@SuppressWarnings("ConfigurationProperties")` to the class.
* Make `SolaceExtendedBindingProperties.DEFAULTS_PREFIX` a private-protected constant and use it as the prefix for the `@ConfigurationProperties` value. i.e. `@ConfigurationProperties(DEFAULTS_PREFIX + ".consumer")`

**Review by Nephery — CHANGES_REQUESTED** [review]

File: solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/pom.xml [hunk 3]
```diff
@@ -44,6 +44,13 @@
             <artifactId>spring-cloud-stream</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <version>2.6.1</version>
```
[the remark this answers is not in the document]
Comment (carolmorneau) [reply solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/pom.xml]: Done

File: solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceProducerProperties.java (hunk 1 above)
Comment (carolmorneau) [reply solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceProducerProperties.java]: Done

File: solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceConsumerProperties.java (hunk 2 above)
Comment (carolmorneau) [reply solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/properties/SolaceConsumerProperties.java]: Done

**Review by Nephery — APPROVED** [review]

**Pull request merged**